{"hash":"782414107f461c86c0093807ec9a7009f522142f","data":{"article":{"id":"07a60d2a81a5f467d244f11ba37c03b9","title":"Named Callbacks in JavaScript","date":"August 06, 2013","categories":["Development"],"permalink":"named-callbacks-in-javascript","content":"<p>You've been taught bad JavaScript for years and years.  What do I mean?  I'm sure you've written code like this more than once in your career:</p>\n<pre class=\"theme:github lang:js decode:true\">someMethod(function (){\n     console.log(\"doing stuff...\");\n});</pre>\n<p>So what's wrong with that?  It runs and doesn't error.  Heck, even JSLint won't complain about it.  But what happens something goes wrong inside of your callback?  Most debuggers will list the callback as an \"anonymous method\" in the call stack.  Not very helpful is it?  Isn't there something simple we can do make our callbacks easier to read in a call stack?</p>\n<p><strong>NAME YOUR CALLBACKS.</strong></p>\n<p>Take the example above, and let's name the callback.</p>\n<pre class=\"lang:js decode:true\">someMethod(function myCallback(){\n     console.log(\"doing stuff...\");\n});</pre>\n<p>Simple!  Let's imagine doing a jQuery AJAX call with this approach (and yes, I'm ignore the fact that you should use promises -- which are WAY better):</p>\n<pre class=\"lang:default decode:true\">$.ajax({\n  url: \"/myUrl\",\n  success: function onAjaxSuccess (data){\n           },\n  error: function onAjaxError (error) {\n           }\n});</pre>\n<p>You're welcome :)</p>\n"}},"context":{}}